Packaging CBZ and commandline arguments

matrix07012 opened the pull request:
I added the ability to package chapters into CBZ, commandline arguments, retrying downloads, tweaked README and bumped the version to v0.3, because I think that new features warrant it.

Commit:
Add ability to package chapters into cbz

Commit:
Command line arguments and image download retry

Commit:
Modified readme and bumped up the version

matrix07012 committed:
Merge pull request #1 from frozenpandaman/master

Merge with main

matrix07012 committed:
Merge branch 'cbz_and_argparse' into master

matrix07012 committed:
Merge pull request #2 from matrix07012/master

Update

matrix07012 committed:
Merge remote-tracking branch 'upstream/master'

orbea commented on mangadex-dl.py:
```diff
@@ -132,12 +135,17 @@ def dl(manga_id, lang_code, zip_up, tld="org"):
+						with open(outfile, 'wb') as f:
+							f.write(r.content)
+				except:
+					print("Encountered an error when downloading. Retrying...")
```
Maybe it would be better to keep the old error, but only print it when bailing out after the last retry?

matrix07012 committed:
Merge pull request #5 from frozenpandaman/master

Bring up to date

frozenpandaman commented:
Hey @matrix07012! I think this functionality would be very cool to have, and making it (and the current language code thing) available via an optional argument with argparse is a great suggestion. Not sure if you've been following the news with the site in the past few months & the leak/hiatus/downtime, but there's an all-new API for the soon-to-be-newly-refreshed site; I've now updated the script (huge revamp) to work with it – and it does, even with the site frontend still down. If you'd be interested in fixing up this PR (I know it's way old now, apologies!) to incorporate this functionality I'd be keen to include it, otherwise I might get around to it some point on my own. :D

Thanks!

Commit:
Merged with upstream

matrix07012 committed:
Merge pull request #7 from matrix07012/frozenpandaman-master

matrix07012 committed:
Merge pull request #8 from matrix07012/master

Update

matrix07012 commented:
Thanks for a reminder. I updated the branch and also fixed a bug that I discovered where the files in the CBZ retained their entire directory structure.

frozenpandaman commented:
Thanks, awesome! I'll merge this into a new branch to do some final touchups/changes (I think just having it be an argument rather than a prompt every run will be best, and want to update the readme, etc.) before merging it into master!
